testsuite/smoke-test/src/operational_tooling.rs

```diff
@@ -257,6 +280,134 @@ fn test_extract_public_key() {
     assert_eq!(key_from_storage, key_from_file);
 }
 
+#[test]
+fn test_extract_peer_from_storage() {
+    let (mut env, op_tool, backend, _) = launch_swarm_with_op_tool_and_backend(1, 0);
+    env.setup_vfn_swarm();
+
+    // Check Validator
+    let (config, _) = load_node_config(&env.validator_swarm, 0);
+    let map = op_tool
+        .extract_peer_from_storage(VALIDATOR_NETWORK_KEY, &backend)
+        .unwrap();
+    let network_config = config.validator_network.unwrap();
+    let expected_peer_id = network_config.peer_id();
+    let expected_public_key = network_config.identity_key().public_key();
+    let (peer_id, peer) = map.iter().next().unwrap();
+    assert_eq!(expected_public_key, *peer.keys.iter().next().unwrap());
+    assert_eq!(expected_peer_id, *peer_id);
+
+    // Check VFN now
+    let (config, _) = load_node_config(&env.vfn_swarm.unwrap(), 0);
```

Comment:
This line is the only changed line from main, and it's because there were testing changes put in after the release cut.